### PR TITLE
JavaScript Todo App Project

### PR DESCRIPTION
**Pull request by adaorachi**
This is the first draft of this project and the following features were implemented;

- [ ] A user has the ability to create projects.

- [ ] A user can create tasks for each project and a task can only belong to a project.

- [ ] A user has the option of adding icons to identify a project when creating them.

- [ ] A user can get a task completed by clicking on the list card.

- [ ] A user can view all tasks belonging to a project from the project card.

- [ ] A user can modify, edit and delete task.

Kindly make a review on it. We hope to get feedback soon.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "js-todo-app",
+  "version": "1.0.0",
+  "description": "A to-do list app built with JavaScript",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack",
+    "watch": "webpack --watch",
+    "server": "webpack-dev-server --open"
```
Comment (mcrd25) [inline package.json]: [OPTIONAL]

This script does not run because your dependency `webpack-dev-server` is not listed on dependencies and is not installed with `npm install`

Also consider having this script be `start`

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 👀 

Hey @adaorachi! Hey @mcastorena0316! 👋 

Excellent job! Love the effort! There is one requirement that needs to be addressed before you can get the 👍 though.

#### Required Changes
- You ought to encapsulate your `ToDo` into its own object (Either factory or contractor) with `title`, `priority`, `dueDate`, `description` properties. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/todo_requirement.png) from Odin.


Hope this review helps.

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

File: src/logic.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    let checkedButton = ' ';
+
+    const formButtons = document.getElementsByName('priorityRadios');
+    formButtons.forEach((button) => {
+      if (button.checked) {
+        checkedButton = button.value;
+      }
+    });
+
+    const project = Task(tName, tDate, tDescription, checkedButton, tID, completedTask);
```
Comment (shloch) [inline src/logic.js]: Before creating new task, please apply some controls to ensure it's rightfully filled, maybe minimum number of characters, or simply check it's not EMPTY

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @adaorachi , @mcastorena0316 

Great job implementing the changes of Previous TSE.
I like your design and the icons, it's really pretty.

### [changes required]

Your app works very well....i just have one more useful security requirement

- imagine when creating a new task, i just validate the empty form, it will generate the error below :
Since date-field is empty, we get the console error below.
And from that error, you can't add further tasks...it blocks the entire app.....you only have to clear the LOCAL STORAGE before you can start again
Screen Shot 2020-04-11 at 4 16 26 PM

- imagine also that when creating a new task, i enter just a correct date, but no task-title nor task-description, i get the screenshot below
Screen Shot 2020-04-11 at 4 20 56 PM

### Basically, apply minimum controls, and you'll solve all these...HAPPY CODING

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @adaorachi @mcastorena0316 kudos for working on this and making changes as suggested the UI looks really nice an everything works fine now, please merge and move on to **this is approved** :+1: 

**RAPHAEL NORIODE** :world_map:
